ygen/protogen.go

```diff
@@ -101,14 +101,21 @@ message {{ .Name }} {
 	}
 )
 
-// writeProtoHeader outputs the header for a proto3 generated file. The package
-// name, base for the generated packages, files for which the protobuf was
-// generated, and the paths used for include are supplied as arguments. In
-// addition, whether path compression is being performed, and the calling
-// binary name are provided as arguments. Returns a string containing the
-// generated header code.
-func writeProtoHeader(pkg, basePkg, baseImportPath string, imports, yangFiles, includePaths []string, compressPaths bool, caller string) (string, error) {
-
+// writeProto3Header outputs the header for a proto3 generated file. Takes
+// the following arguments:
+//  - pkg - the name of the protobuf package being written.
+//  - base - the base for the package, such that the total package name is
+//    represented as base.pkg.
+//  - imports - the other protobuf packages that should be imported by the
+//    protobuf package.
+//  - yangFiles - the set of files that were input to the library.
+//  - includePaths - the set of include paths that were input to the library.
+//  - compressPaths - a bool indicating whether path compression should
+//    be used.
+//  - caller - the name of the calling binary - included in the package
+//    header.
+// It returns a string containing the generated header code.
+func writeProto3Header(pkg, basePkg, baseImportPath string, imports, yangFiles, includePaths []string, compressPaths bool, caller string) (string, error) {
```

Comment:
(optional) sorry, should have mentioned in the first round: the list of params here is very long and has a bunch of fields of the same type which are easy for the caller to mix up - consider defining a struct.

Reply:
Done.